### PR TITLE
[DOC-10608] Document new log flushing behavior under systemd

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-10608] Document new log flushing behavior under systemd

### CODE REVIEW

File: src/current/v24.3/configure-logs.md [hunk 1]
```diff
@@ -637,6 +637,8 @@ With this logging configuration:
 - If at any point the accumulated message size of buffer flushes (as triggered by reaching either the configured `max-staleness` or `flush-trigger-size` value) exceeds `100MiB` (the `max-buffer-size` setting), all new incoming log messages received are dropped until the accumulated message size in the buffer once more falls below this value.
 - For the `HEALTH` [log channel]({% link {{ page.version.version }}/logging-overview.md %}#logging-channels) only, override the `file-defaults` value of `20s` for `max-staleness`, instead flushing messages to the log file within up to `2s`.
 
+{% include_cached new-in.html version="v24.3" %}If CockroachDB is managed using `systemd` and log buffering is enabled, logs are flushed automatically when CockroachDB starts.
```
Comment (jaylim-crl) [inline src/current/v24.3/configure-logs.md:640]: This doesn't reflect the intention of the PR in https://github.com/cockroachdb/cockroach/pull/125971/files. Previously, without the change, logs are flushed regardless whenever CockroachDB starts. Now, with that change, logs are only flushed when CockroachDB is managed using systemd.

Comment (mdlinville) [inline src/current/v24.3/configure-logs.md:640]:
```suggestion
{% include_cached new-in.html version="v24.3" %}Logs are flushed automatically when CockroachDB starts only if CockroachDB is managed using `systemd` and log buffering is enabled.
```
Reworded this a bit to hopefully address the feedback.

File: src/current/v24.3/configure-logs.md [hunk 2]
```diff
@@ -637,6 +637,8 @@ With this logging configuration:
 - If at any point the accumulated message size of buffer flushes (as triggered by reaching either the configured `max-staleness` or `flush-trigger-size` value) exceeds `100MiB` (the `max-buffer-size` setting), all new incoming log messages received are dropped until the accumulated message size in the buffer once more falls below this value.
 - For the `HEALTH` [log channel]({% link {{ page.version.version }}/logging-overview.md %}#logging-channels) only, override the `file-defaults` value of `20s` for `max-staleness`, instead flushing messages to the log file within up to `2s`.
 
+{% include_cached new-in.html version="v24.3" %}When log buffering is enabled, logs are flushed automatically when CockroachDB starts only if CockroachDB is managed using `systemd`.
```
Comment (jaylim-crl) [inline src/current/v24.3/configure-logs.md:640]: nit: ... logs are automatically flushed upon CockroachDB startup only if it is managed by `systemd` ... seems to flow better, but no strong preferences here :)

**Review by jaylim-crl — APPROVED** [review]
LGTM. Thanks for the change!

File: src/current/v24.3/configure-logs.md (hunk 2 above)
Comment (mdlinville) [inline src/current/v24.3/configure-logs.md:640]:
```suggestion
{% include_cached new-in.html version="v24.3" %}Logs are flushed automatically upon CockroachDB startup only if CockroachDB is managed using `systemd`.
```

**Pull request merged**